src/main/scala/scalismo/faces/sampling/face/proposals/ImageCenteredProposal.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright University of Basel, Graphics and Vision Research Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package scalismo.faces.sampling.face.proposals
+
+import scalismo.faces.landmarks.TLMSLandmark2D
+import scalismo.faces.parameters.RenderParameter
+import scalismo.faces.sampling.face.ParametricLandmarksRenderer
+import scalismo.geometry.{Point, Vector}
+import scalismo.sampling.{ProposalGenerator, TransitionProbability}
+
+/**
+  * proposal which keeps the image location of a selected landmark at a constant position, uses shifts of the principle point to keep the position fixed.
+  * Compensates translation of the wrapped proposal
+  *
+  * @param proposal wrapped proposal, translation introduced by it will be compensated
+  * @param lmId id of landmark to keep fixed (must be known by renderer)
+  * @param lmRenderer parametric landmarks renderer to find image position of landmark
+  */
+class ImageCenteredProposal(val proposal: ProposalGenerator[RenderParameter] with TransitionProbability[RenderParameter],
+                            val lmId: String,
+                            val lmRenderer: ParametricLandmarksRenderer)
+  extends ProposalGenerator[RenderParameter] with TransitionProbability[RenderParameter] {
+
+  override def propose(current: RenderParameter): RenderParameter = {
+    // render the landmark's image location before application of the proposal
+    // keep track of the landmark position
+    val lmCurrent: TLMSLandmark2D = lmRenderer.renderLandmark(lmId, current).get // get is safe, checked during construction
+
+    // apply proposal
+    val propSample = proposal.propose(current)
+
+    // render the landmark after application
+    val lmProposal: TLMSLandmark2D = lmRenderer.renderLandmark(lmId, propSample).get
+
+    // shift back (compensate translation in image)
+    val shift = lmCurrent.point - lmProposal.point
+    // strange shifting in y direction
```

Comment:
Fast draft of better comment..... "The special shifting originates from that the principle point lives in a different coordinate system than the landmark which lives in the image coordinates. The y direction is inverted, hence the negation along this axis. Further the full image maps to values from -1 to 1 in both direction, that is the reason for the factor of 2."